### PR TITLE
{compiler}[GCCcore/9.3.0] Add NVHPC v20.9, v20.11 and default to accept EULA

### PR DESCRIPTION
**Pull request by AndiH**
This pull request adds NVHPC 20.11, starting off of #11694.

Additionally, I got approval from @samcmill to default-accept the EULA and have, hence, uncommented the `source_urls` definition. I added a comment relating to the EULA, though.

### CODE REVIEW

**Review by lexming — COMMENTED** [review]
@AndiH could you also add the `source_urls` and comment about the EULA in v20.11 to the easyconfig of v20.9?
I also have a minor suggestion to reduce duplicate code in the easyconfig and make it more readable. Please see below

**Comment by boegel** [comment]
@AndiH W.r.t. the EULA, we could enhance the NVHPC easyblock to use the `EasyBlock.check_accepted_eula` method, see https://github.com/easybuilders/easybuild-framework/pull/3535?

**Comment by AndiH** [comment]
@boegel We could! But since we got the general ok, I'd suggest to rather always accept it!

**Comment by verdurin** [comment]
@AndiH any update on the suggestions from @lexming ?

**Comment by AndiH** [comment]
@verdurin @lexming I think it makes it less readable (because more indirection), but easier to update in the future. I'll gladly implement. Give me a few days, though; I'm currently officially on parental leave :) .

**Comment by AndiH** [comment]
I've added the following changes

* implemented @lexming's suggestions regarding restructuring of source filenames using `local_tarball_tmpl`
* backported these changes plus the EULA default-accept to NVHPC 20.7

File: easybuild/easyconfigs/n/NVHPC/NVHPC-20.11.eb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+name = 'NVHPC'
+version = '20.11'
+
+homepage = 'https://developer.nvidia.com/hpc-sdk/'
+description = """C, C++ and Fortran compilers included with the NVIDIA HPC SDK (previously: PGI)"""
+
+toolchain = SYSTEM
+
+# By downloading, you accept the HPC SDK Software License Agreement (https://docs.nvidia.com/hpc-sdk/eula/index.html)
```
Comment (lexming) [inline easybuild/easyconfigs/n/NVHPC/NVHPC-20.11.eb:9]:
```suggestion
# By downloading, you accept the HPC SDK Software License Agreement (https://docs.nvidia.com/hpc-sdk/eula/index.html)
accept_eula = True
```

File: easybuild/easyconfigs/n/NVHPC/NVHPC-20.9.eb [hunk 2]
```diff
@@ -0,0 +1,68 @@
+name = 'NVHPC'
+version = '20.9'
+
+homepage = 'https://developer.nvidia.com/hpc-sdk/'
+description = """C, C++ and Fortran compilers included with the NVIDIA HPC SDK (previously: PGI)"""
+
+toolchain = SYSTEM
+
+# By downloading, you accept the HPC SDK Software License Agreement (https://docs.nvidia.com/hpc-sdk/eula/index.html)
```
Comment (lexming) [inline easybuild/easyconfigs/n/NVHPC/NVHPC-20.9.eb:9]:
```suggestion
# By downloading, you accept the HPC SDK Software License Agreement (https://docs.nvidia.com/hpc-sdk/eula/index.html)
accept_eula = True
```

**Review by lexming — CHANGES_REQUESTED** [review]
@AndiH thanks for updating the PR. I made the changes requested by @boegel to the easyblock of NVHPC (see https://github.com/easybuilders/easybuild-easyblocks/pull/2311) to better handle the acceptance of the EULA.

Since you already checked that the EULA of NVHPC can be automatically accepted by just downloading the sources, we can add `accept_eula = True` to these easyconfigs to automatize the process. See below

**Comment by AndiH** [comment]
Oh, sorry, I didn't recognize there was additional work for me to be done :). It's done now!

**Comment by verdurin** [comment]
Looks fine.

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node106.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/f8adb4fa839043d16d7df4def48a4fb0 for a full test report.

**Review by lexming — APPROVED** [review]
@AndiH thanks for updating the PR!
@verdurin your test failed because `NVHPC` requires some `--cuda-compute-capabilities` to be set
LGTM

**Comment by lexming** [comment]
Going in, thanks @AndiH !

**Pull request merged**